lib/authorizer/digest.js

```diff
@@ -140,7 +140,8 @@ _extractField = function (string, regexp) {
  */
 function _getDigestAuthHeader (headers) {
     return headers.find(function (property) {
-        return (property.key.toLowerCase() === WWW_AUTHENTICATE) && (_.startsWith(property.value, DIGEST_PREFIX));
+        return (property.key.toLowerCase() === WWW_AUTHENTICATE) &&
+        (_.startsWith(property.value.toString().toLowerCase(), DIGEST_PREFIX.toLowerCase()));
```

Comment:
converting this to String to make sure it's not break for non string header values